src/sidebar/components/MarkdownView.tsx

```diff
@@ -35,6 +41,12 @@ export default function MarkdownView({
     });
   }, [markdown]);
 
+  useEffect(() => {
+    if (hasMentions) {
```

Comment:
If a user creates an annotation with only invalid mentions, the client will wrap the mentions in tags when submitting the annotation to the server, and these tags will contain `data-userid` attributes. These tags match the styling criteria for valid mentions. When the server receives the annotation it will not find matches and hence the `mentions` that come back to the client will be empty. `hasMentions` will end up being false and so `renderMentionsTag` won't run. The invalid mentions will not be modified and will end up being styled as _valid_ mentions.

Reply:
Oh, good catch! This is a scenario I didn't test, nor added unit tests for it.